legend-engine-xt-graphQL-query/src/test/java/org/finos/legend/engine/query/graphQL/api/test/TestGraphQLAPI.java

```diff
@@ -405,6 +406,59 @@ public void testCacheUsed() throws Exception
 
     }
 
+    @Test
+    public void testCachingUsingNestedSelectionSets() throws Exception
```

Comment:
This tests the cache is missed, so name should be testCacheMissOnNestedSelectionSets or something like this.

Also we should have a test for cache hit using nested selection sets.